ethereumj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1328,8 +1333,8 @@ public void play(Program program) {
         }
     }
 
-    public static void setVmHook(VMHook vmHook) {
-        VM.vmHook = vmHook;
+    public static void setVmHookFactory(VMHookFactory vmHookFactory) {
```

Comment:
Let's keep old `VMHook` behavior for backward compatibility and mark it as deprecated with pointing to new alternative with detailed description of why client should use new feature